buildSrc/src/main/java/com/synopsys/integration/detect/docs/DocGenerator.kt

```diff
@@ -52,25 +53,57 @@ open class GenerateDocsTask : DefaultTask() {
         FileUtils.copyDirectory(project.file("docs/static"), outputDir)
 
         createFromFreemarker(templateProvider, outputDir, "exit-codes", ExitCodePage(helpJson.exitCodes))
-        createFromFreemarker(templateProvider, outputDir, "index", IndexPage(project.version.toString()))
 
         handleDetectors(templateProvider, outputDir, helpJson)
         handleProperties(templateProvider, outputDir, helpJson)
+        handleContent(outputDir, templateProvider)
+    }
+
+    private fun handleContent(outputDir: File, templateProvider: TemplateProvider) {
+        val helpContentTerms = Terms()
+        helpContentTerms.termMap.put("program_version", project.version.toString())
+        project.file("docs/templates/content").walkTopDown().forEach {
+            if (it.canonicalPath.endsWith((".ftl"))) {
+                createContentMarkdownFromTemplate(it, outputDir, templateProvider, helpContentTerms)
+            }
+        }
+    }
+
+    private fun createContentMarkdownFromTemplate(templateFile: File, baseOutputDir: File, templateProvider: TemplateProvider, helpContentTerms: Terms) {
+        val helpContentTemplateRelativePath = deriveTemplateRelativePath(templateFile)
+        val outputFile = deriveOutputFileForContentTemplate(helpContentTemplateRelativePath, baseOutputDir)
+        println("Generating markdown from template file: ${helpContentTemplateRelativePath} --> ${outputFile.canonicalPath}")
+        createFromFreemarker(templateProvider, helpContentTemplateRelativePath, outputFile, helpContentTerms.termMap)
+    }
+
+    private fun deriveOutputFileForContentTemplate(helpContentTemplateFileRelativePath: String, baseOutputDir: File): File {
```

Comment:
I am not a fan of string manipulation for handling file paths. Not sure I have a great alternative at first glance but it gives me pause.

Reply:
I've replaced the string manipulation with methods from the File class

Reply:
I also now use shared variables... good idea.